openedx/core/djangoapps/credit/tests/fake_update_min_grade.py

```diff
@@ -0,0 +1,70 @@
+"""
+Fake page for updating the minimum grade course requirement status for a user
+"""
+from django.contrib.auth.decorators import login_required
+from django.core.context_processors import csrf
+from django.utils.decorators import method_decorator
+from django.views.generic.base import View
+
+from openedx.core.djangoapps.credit.models import CreditCourse, CreditRequirement, CreditRequirementStatus
+
+from edxmako.shortcuts import render_to_response
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+from opaque_keys.edx.locations import SlashSeparatedCourseKey
+
+from xmodule.modulestore.django import modulestore
+
+
+class UpdateMinRequirementFakeView(View):
+    """
+    Fake view for updating minimum grade status in
+    credit eligibility status table.
+    """
+
+    @method_decorator(login_required)
+    def get(self, request):
+        """
+        Render a fake page for updating the minimum grade course requirement
+        status for a user.
+        """
+        context = {
+            'csrf_token': unicode(csrf(request)['csrf_token'])
+        }
+
+        return render_to_response("credit/fake_credit_eligibility_status.html", context)
+
+    @method_decorator(login_required)
+    def post(self, request):
+        """
+        After validating all conditions it will update minimum grade
+        status in credit eligibility requirements status table.
+        """
+        context = {}
+        course_key, error = None, None
+        course_id = request.POST.get("txt_coursekey")
+        try:
+            course_key = CourseKey.from_string(course_id)
+        except InvalidKeyError:
+            try:
```

Comment:
No need to add another try catch. `course_key = CourseKey.from_string(course_id)` is enough to get course key.